checker.cpp

```diff
@@ -3,44 +3,65 @@
 #include <unistd.h>
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
+
+void Sleep()
+{
+     for (int i = 0; i < 6; i++)
     {
       cout << "\r* " << flush;
       sleep(1);
       cout << "\r *" << flush;
       sleep(1);
     }
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
+int TemperatureOfOk(float temperature)
+{
+    if(temperature > 102 || temperature <95)
     {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+        cout<<"Temperature critical!\n";
+        Sleep();
+        return 0;
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
+    return 1;
+    
+}
+
+int PulseRate(float pulseRate)
+{
+    if(pulseRate < 60 || pulseRate > 100)
     {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+        cout<<"Pulse Rate is out of range!\n";
+        Sleep();
+        return 0;
+    }
+    return 1;
+}
+
+int Spo2(float spo2)
+{
+    if(spo2 < 90)
+    {
+        cout << "Oxygen Saturation out of range!\n";
+        Sleep();
+        return 0;
+    }
+    return 1;
+}
+
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+  if(!TemperatureOfOk(temperature) || !PulseRate(pulseRate) || !Spo2(spo2))
```

Comment:
You can express this in a concise way, by directly returning...
 Like `return ! TemperatureOfOk(...) || !...`